src/Symfony/Component/HttpClient/RetryableHttpClient.php

```diff
@@ -46,6 +46,10 @@ public function __construct(HttpClientInterface $client, ?RetryStrategyInterface
         $this->strategy = $strategy ?? new GenericRetryStrategy();
         $this->maxRetries = $maxRetries;
         $this->logger = $logger;
+
+        if (null !== $logger && $this->client instanceof LoggerAwareInterface) {
+            $this->client->setLogger($logger);
```

Comment:
I am not convinced that a decorator should actually modify the decorated HTTP client this way. IMO setting the logger is the user's responsibility when constructing the client.